lib/puma/systemd.rb

```diff
@@ -29,6 +29,29 @@ def start_watchdog
       end
     end
 
+    def start_status_loop
+      instance = self
+      sleep_time = 1.0
+      log "Sending status to systemd every #{sleep_time.round(1)} sec"
+
+      Thread.new do
```

Comment:
Rather than create threads ourselves, `puma` should provide an API that a puma plugin can use to manage this process.

Reply:
That sounds like a good idea because it should also be used to stop. Just like the thread for the watchdog. It may be more important in the test suite, but right now it may be leaking the thread.

And there is actually for plugins some code for plugins with threads:
https://github.com/puma/puma/blob/eb59ed00d71f66766d75d7c40f671d7016b9d5ae/lib/puma/plugin.rb#L60-L71

Perhaps that should be utilized instead?

Reply:
TIL 😆 Yup that's fine too.